ddtrace/settings/_otel_remapper.py

```diff
@@ -0,0 +1,187 @@
+                    dd_tags.append("{}:{}".format(key, value))
+                else:
+                    remaining_tags.append("{}:{}".format(key, value))
+    except Exception:
+        log.warning("DDTRACE failed to read OTEL_RESOURCE_ATTRIBUTES. This value is misformatted: %s", otel_value)
+
+    if len(otel_user_tag_dict.items()) > 10:
+        log.warning(
+            "To preserve metrics cardinality, only the following first 10 tags have been processed %s. "
+            "The following tags were not ingested: %s",
+            dd_tags,
+            remaining_tags,
+        )
+    return ",".join(dd_tags)
+
+
+def _remap_otel_sdk_config(otel_value):
+    """Remaps the otel sdk config to ddtrace sdk config"""
+    if otel_value == "false":
+        return "True"
+    elif otel_value == "true":
+        return "False"
+    else:
+        log.warning("OTEL_SDK_DISABLED='%s'  is not supported", otel_value)
+        return otel_value
+
+
+def _remap_default(otel_value):
+    """Remaps the otel default value to ddtrace default value"""
+    return otel_value
+
+
+ENV_VAR_MAPPINGS = {
+    "OTEL_SERVICE_NAME": ("DD_SERVICE", _remap_default),
+    "OTEL_LOG_LEVEL": ("DD_TRACE_DEBUG", _remap_otel_log_level),
+    "OTEL_PROPAGATORS": ("DD_TRACE_PROPAGATION_STYLE", _remap_otel_propagators),
+    "OTEL_TRACES_SAMPLER": ("DD_TRACE_SAMPLE_RATE", _remap_traces_sampler),
+    "OTEL_TRACES_EXPORTER": ("DD_TRACE_ENABLED", _remap_traces_exporter),
+    "OTEL_METRICS_EXPORTER": ("DD_RUNTIME_METRICS_ENABLED", _remap_metrics_exporter),
+    "OTEL_LOGS_EXPORTER": ("", _remap_logs_exporter),  # Does not set a DDTRACE environment variable.
+    "OTEL_RESOURCE_ATTRIBUTES": ("DD_TAGS", _remap_otel_tags),
+    "OTEL_SDK_DISABLED": ("DD_TRACE_OTEL_ENABLED", _remap_otel_sdk_config),
+}
+
+
+def otel_remapping():
+    """Checks for the existence of both OTEL and Datadog tracer environment variables and remaps them accordingly.
+    Datadog Environment variables take precedence over OTEL, but if there isn't a Datadog value present,
+    then OTEL values take their place.
+    """
+    user_envs = {key.upper(): value for key, value in os.environ.items()}
+
+    for otel_env, otel_value in user_envs.items():
+        if otel_env not in ENV_VAR_MAPPINGS.keys():
+            continue
+
+        dd_env, otel_config_remapper = ENV_VAR_MAPPINGS[otel_env]
+        if dd_env in user_envs:
+            log.debug(
+                "Datadog configuration %s is already set. OpenTelemetry configuration will be ignored: %s=%s",
+                dd_env,
+                otel_env,
+                otel_value,
+            )
+            continue
+
+        if otel_env not in ["OTEL_RESOURCE_ATTRIBUTES", "OTEL_SERVICE_NAME"]:
+            # Resource attributes and service name are case-insensitive
+            otel_value = otel_value.lower()
+
+        mapped_value = otel_config_remapper(otel_value)
+        if mapped_value != "":
+            os.environ[dd_env] = mapped_value
```

Comment:
should we be checking for `dd_env` being `""` since we have one mapping which maps to `""`, wouldn't that mean for `OTEL_LOGS_EXPORTER` we are setting `os.environ[""] = value` ?

Reply:
I guess this is handled by the remapper, but I guess I still wonder why we are trying to remap it if we aren't going to use it?

Reply:
alternative, instead of setting up a remapping for it, we can just add a `if user_envs.get("OTEL_LOGS_EXPORTER", "none") != "none": log.warning(...)` to the start of this remapping function.

it might make the code for this specific env variable more clear.

Reply:
Hey! Small implementation detail, but: `_remap_logs_exporter` always returns an empty string. It's not a remapper per say, but it's used to detect invalid inputs and log a warning. The empty string env var will not be set. Let me know if that's alright

I renamed _remap_logs_exporter to _validate_logs_exporter to make this more clear

Reply:
@brettlangdon We're also planning to add support for the logs exporter in the near future (so this will remap to a value later on)

Reply:
ok, knowing that we will add support for it makes sense to keep a remapper setup (I am ok with the remap name btw)